lib/vector-vrl/functions/src/set_semantic_meaning.rs

```diff
@@ -68,8 +68,10 @@ impl Function for SetSemanticMeaning {
             .expect("meaning not bytes")
             .into_owned();
 
-        // Semantic meaning can only be assigned to external fields.
-        if !query.is_external() {
+        let path = if let Some(path) = query.external_path() {
+            path
+        } else {
+            // Semantic meaning can only be assigned to external fields.
             let mut labels = vec![Label::primary(
                 "the target of this semantic meaning is non-external",
```

Comment:
Slight aside, but I think it would be better for the error message to say it is not a [path](https://vector.dev/docs/reference/vrl/expressions/#path), in line with what the docs call it..

Reply:
It is still a path, it's just a path that points to a local variable which isn't valid for semantic meaning. I can still update the error message though to make it more clear.